[FIX] New/stepper

larseirikhansen opened the pull request:
Vet ikke helt hva som gjenstår her, så tenker det hadde vært bra med litt feedback på APIet og funksjonaliteten her. Gidder du kikke når du har tid @andnorda? :D 

component
- Stepper.tsx
- Step.tsx
- Stepper.css
sb
- stepper.stories.mdx (intro)
- stepper.stories.tsx

KenAJoh committed:
satt opp stepper react

KenAJoh committed:
satt opp stepper css

KenAJoh committed:
oppdatert react-kode

KenAJoh committed:
førsteutkast styling

larseirikhansen committed:
♿ aria-label added to icons

larseirikhansen committed:
Stepper spacing and last-child focus

larseirikhansen committed:
Merge remote-tracking branch 'origin/master' into new/stepper

larseirikhansen committed:
Merge remote-tracking branch 'origin/master' into new/stepper

larseirikhansen committed:
add aria-labelledby

larseirikhansen committed:
add aria-labelledby to sb

larseirikhansen committed:
fix spelling mistake

larseirikhansen committed:
Remove 'finished' step api-option

larseirikhansen committed:
refactor Stepper-children jsx

larseirikhansen committed:
create Stepper example for sb

larseirikhansen committed:
WIP: Stepper mdx with practical props examples

larseirikhansen committed:
Merge remote-tracking branch 'origin/master' into new/stepper

larseirikhansen committed:
intro mdx disabled state

larseirikhansen committed:
ds intro overridable tags example

larseirikhansen committed:
Merge remote-tracking branch 'origin/master' into new/stepper

larseirikhansen committed:
add flex gap and fix last-child margin

larseirikhansen committed:
step-line is now pseudo-element

larseirikhansen committed:
cleanup css

larseirikhansen committed:
Merge remote-tracking branch 'origin/master' into new/stepper

larseirikhansen committed:
last child no margin-bottom

larseirikhansen committed:
remove disabled state

larseirikhansen committed:
update storybook

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+  flex-shrink: 0;
+}
+.navds-stepper__step-wrapper:last-child > .navds-stepper__step {
+  min-height: auto;
```
<img width="255" alt="Screenshot 2022-03-11 at 11 21 45" src="https://user-images.githubusercontent.com/26967723/157848992-763c70ac-2eb1-47f8-81d8-31e0e39813fb.png">

Hvert steg bør kanskje ha en min-høyde slik at siste steg ikke blir squishet helt. Eller er dette en del av designet?

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+}
+
+/* Step */
+
```
```suggestion
```

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+}
+
+/* Step focus */
+
```
```suggestion
```

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+}
+
+/* Line */
+
```
```suggestion
```

KenAJoh commented on @navikt/core/react/src/stepper/Step.tsx:
```diff
@@ -0,0 +1,61 @@
+import { StepperContext } from ".";
+import { Label, OverridableComponent } from "..";
+
+export interface StepperStepProps
```
![Screenshot 2022-03-11 at 11 37 28](https://user-images.githubusercontent.com/26967723/157852188-0c30ffd4-1947-42fd-98ac-b479c2776cef.png)

Ser litt forskjellige variasjoner for "paginering" av stepper. Er dette versjonene som er bestemt nå?

KenAJoh replied:
Om så vil kanskje next/prev fungere som egne knapper, eks: `<Stepper.Previous/> / <Stepper.Next/>`

andnorda replied:
Jeg stemmer for at vi implementerer den funksjonaliteten i en annen PR.

KenAJoh commented on @navikt/core/react/src/stepper/Step.tsx:
```diff
@@ -0,0 +1,61 @@
+          rest?.onClick?.(e);
+        }}
+      >
+        <Label className="navds-stepper__step-number">
```
```suggestion
        <Label className="navds-stepper__step-number" as="span">
```
Endrer `<p>` -> `<span>`

KenAJoh commented on @navikt/core/react/src/stepper/Stepper.tsx:
```diff
@@ -0,0 +1,86 @@
+  /**
+   * ID of element that labels the Stepper-element
+   */
+  "aria-labelledby"?: string;
```
Trenger sikkert ikke definere denne i props da vi får den fra `React.HTMLAttributes<HTMLOListElement>`.

larseirikhansen replied:
done!

KenAJoh commented on @navikt/core/react/src/stepper/Stepper.tsx:
```diff
@@ -0,0 +1,86 @@
+        >
+          {React.Children.map(children, (step, index) => {
+            return (
+              <li className={cl("navds-stepper__step-wrapper")} key={index}>
```
Er ikke 100% på denne, men tror `key={index}` kan brekke hvis man endrer antall keys dynamisk ved bruk da index blir endret.

larseirikhansen replied:
Done!

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+  flex-direction: column;
+  padding: 0;
+  margin: 0;
+  gap: 0.5rem;
```
```suggestion
  gap: var(--navds-spacing-2);
```

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+  width: 28px;
+  height: 28px;
+  margin: 0;
+  margin-right: 0.75rem;
```
```suggestion
  margin-right: var(--navds-spacing-3);
```

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+  height: 28px;
+  margin: 0;
+  margin-right: 0.75rem;
+  border-radius: 50%;
```
```suggestion
  border-radius: var(--navds-border-radius-full);
```

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,89 @@
+  height: calc(100% - 28px - 8px);
+  display: block;
+  border-left: 2px solid var(--navds-step-indicator-color-step-line);
+  border-radius: 4px;
```
```suggestion
  border-radius: var(--navds-border-radius-medium);
```

KenAJoh reviewed (dismissed):
Har kommentert noen småting, men ser veldig bra ut! 💯

KenAJoh committed:
[FIX] Bedre width for search (#1455)

github-actions[bot] committed:
Lerna versionbump [ci skip]

 - @navikt/ds-css@0.16.3

andnorda committed:
[fix] Helptext uu fix (#1407)

* storybook no div in p fix

* move title inside svg

* extra helptext in story

* remove duplicate title

* remove aria-props and refactor

* remove unused style

* simplify styling some

* default help text content to body-short

* cleanup

* tabindex 0 på popover

* flyttet story

* fjernet tabindex

* Update @navikt/core/react/src/help-text/HelpText.tsx

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

Co-authored-by: Ken <ken.aleksander@gmail.com>
Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

github-actions[bot] committed:
Lerna versionbump [ci skip]

 - @navikt/ds-css@0.16.4
 - @navikt/ds-react@0.17.4
 - @navikt/ds-react-internal@0.12.4

larseirikhansen committed:
aria-current to active step

larseirikhansen committed:
styling and color fixes

larseirikhansen committed:
color change line step

KenAJoh committed:
[FIX] Oppdatert ikon-template, SSR-safe uuid (#1458)

* oppdatert ikon-template

* oppdatert pakkversjoner

* fikset pakkeversjoner

* oppdatert ikon-template

* fikset config

github-actions[bot] committed:
Lerna versionbump [ci skip]

 - @navikt/ds-icons@0.8.5
 - @navikt/ds-react@0.17.5
 - @navikt/ds-react-internal@0.12.5

andnorda committed:
[fix] Read more component (#1456)

* scrips yo

* unused import

* stylez

* aria-expanded

* add link to storybook

* oopsie

* read more focus style

* read more collapse animation

github-actions[bot] committed:
Lerna versionbump [ci skip]

 - @navikt/ds-css@0.16.5
 - @navikt/ds-react@0.17.6
 - @navikt/ds-react-internal@0.12.6

larseirikhansen committed:
Update @navikt/core/css/stepper.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
Update @navikt/core/css/stepper.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
Update @navikt/core/css/stepper.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
Update @navikt/core/css/stepper.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
Update @navikt/core/css/stepper.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
Update @navikt/core/css/stepper.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
Update @navikt/core/css/stepper.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
Update @navikt/core/react/src/stepper/Step.tsx

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

larseirikhansen committed:
remove ariaLabelledBy as its in by default

KenAJoh committed:
Merge branch 'master' into new/stepper

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,101 @@
+}
+
+/* Step label */
+
```
```suggestion
```

KenAJoh committed:
Update @navikt/core/css/stepper.css

andnorda commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,101 @@
+  cursor: pointer;
+  flex-shrink: 0;
+}
+.navds-stepper__step-wrapper:last-child > .navds-stepper__step {
```
```suggestion
}

.navds-stepper__step-wrapper:last-child > .navds-stepper__step {
```

andnorda commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,101 @@
+  box-shadow: var(--navds-shadow-focus);
+  outline: none;
+}
+.navds-stepper__step--active {
```
```suggestion
}

.navds-stepper__step--active {
```

andnorda commented on @navikt/core/react/src/stepper/Step.tsx:
```diff
@@ -0,0 +1,62 @@
+        }}
+      >
+        <Label className="navds-stepper__step-number" as="span">
+          {context.activeStep === index ? ` ${index + 1}` : index + 1}
```
Hva skjer her?

andnorda commented on @navikt/core/react/src/stepper/Stepper.tsx:
```diff
@@ -0,0 +1,73 @@
+    { children, className, activeStep, onStepChange = () => {}, ...rest },
+    ref
+  ) => {
+    return (
```
Bruk gjerne oneline-sukker her.

andnorda commented on @navikt/core/react/src/stepper/Step.tsx:
```diff
@@ -0,0 +1,62 @@
+  /**
+   * Text content under indicator
+   */
+  children: React.ReactNode;
```
```suggestion
  children: string;
```

KenAJoh replied:
```suggestion
  children: string;
```

larseirikhansen committed:
anchor tag is default overridable

larseirikhansen committed:
rm space

andnorda commented on @navikt/core/react/src/stepper/Step.tsx:
```diff
@@ -0,0 +1,59 @@
+        }}
+      >
+        <Label className="navds-stepper__step-number" as="span">
+          {context.activeStep === index ? `${index + 1}` : index + 1}
```
```suggestion
          {index + 1}
```

larseirikhansen committed:
default from button to anchor

larseirikhansen committed:
add y-axis click affordance to steps

larseirikhansen committed:
click affordance, flex-align-start and stretching h-line with long label

andnorda reviewed (dismissed):
LGTM

KenAJoh commented on @navikt/core/css/stepper.css:
```diff
@@ -0,0 +1,100 @@
+}
+
+/* Step */
+
```
```suggestion
```

KenAJoh committed:
Update @navikt/core/css/stepper.css
